geco/graphs/utilities.py

```diff
@@ -35,6 +35,73 @@ def edgeweight_properties(graph):
     return max_weight, min_weight, num_of_zero_weights
 
 
+def num_of_simple_cycles(graph):
+    """
+    Finds number of simple cycles
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        Graph to find number of simple cycles in
+
+    Returns
+    -------
+    num_of_simple_cycles: int
+        Number of simple cycles in graph
+
+    Notes
+    -----
+    It was pointed out that this implementation
+    is probably buggy in the undirected case.
+    """
+    if graph.is_directed():
+        return len(list(nx.simple_cycles(graph)))
+    else:
+        graph = nx.DiGraph(graph)
+        all_simple_cycles = list(nx.simple_cycles(graph))
+        all_simple_cycles = filter(lambda x: len(x) > 2, all_simple_cycles)
+        all_simple_cycles = set(map(lambda x: tuple(sorted(x)), all_simple_cycles))
+        return len(list(all_simple_cycles))
+
+
+def find_parallel_edges(graph):
+    """
+    Finds parallel edges (not total edges) between nodes and their amount.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        Graph to find parallel edges in
+
+    Returns
+    -------
+    parallel_edges: list
+        Edge as tuple, followed by number of parallel edges
+
+    Notes
+    -----
+    If the graph is undirected, this implementation finds all parallel
+    edges twice (once from both vertices), then filters out the duplicate.
+    """
+    all_parallel_edges = []
+    for node in graph:
+        for neighbor in graph.neighbors(node):
+            num_of_edges = graph.number_of_edges(node, neighbor)
+            if num_of_edges > 1:
+                all_parallel_edges.append([(node, neighbor), num_of_edges - 1])
```

Comment:
Since the data size isn't dynamic, I think using tuples to hold the edge data makes more sense here.